llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1437,6 +1438,20 @@ void AsmPrinter::emitFunctionBody() {
         MCSymbolRefExpr::create(CurrentFnEnd, OutContext),
         MCSymbolRefExpr::create(CurrentFnSymForSize, OutContext), OutContext);
     OutStreamer->emitELFSize(CurrentFnSym, SizeExp);
+    if (CurrentFnLocalForEH) {
+      // For CHERI exception landing pad tables we need a local alias so
+      // that we can emit non-preemptible relocations with the correct addend.
+      // See https://github.com/CTSRD-CHERI/llvm-project/issues/512.
+      // Note: We add .local so that the symbol is not deleted after assembling
+      // even though the name might start with ".L".
+      OutStreamer->emitSymbolAttribute(CurrentFnLocalForEH, MCSA_Local);
+      OutStreamer->emitSymbolAttribute(CurrentFnLocalForEH,
+                                       MCSA_ELF_TypeFunction);
+      OutStreamer->emitAssignment(
+          CurrentFnLocalForEH,
+          MCSymbolRefExpr::create(CurrentFnSymForSize, OutContext));
+      OutStreamer->emitELFSize(CurrentFnLocalForEH, SizeExp);
```

Comment:
The local and assignment probably shouldn't be under hasDotTypeDotSizeDirective even though that's always the case for us?

Reply:
Yes, I only did that so I can re-use SizeExp, but I can just move that outside the if?

Reply:
The size and type would stay here, but the local and assignment would move out?

Reply:
That would work but then we have .type after the assignment. Not sure it's a problem, but it seems like the wrong order to me.

Reply:
So do it before the if not after?

Reply:
If that bothers you you can expose SizeExp then